### PR TITLE
Fixed issue #18559: Potential 500 error with LimeMailer and PHP8

### PR DESCRIPTION
**Pull request by Shnoulle**
Dev: Use PHPMailer default for all if empty by plugin (or before)
Dev: really needed currently by to only

### CODE REVIEW

**Review by gabrieljenik — APPROVED** [review]
Code looks OK. Haven't tested it.

**Comment by Shnoulle** [comment]
> Who tested it? In production?

me … i add the "Hard to test" tag ;) 

Fix another plugin : https://gitlab.com/SondagesPro/SurveyAccess/surveyChaining/-/commit/bb405f14f3654d548de3b81cfb67accfe7ef72ab

I can send the previous plugin again.

The issue is when a plugin set email to false

File: application/core/LimeMailer.php [hunk 1]
```diff
@@ -515,24 +515,27 @@ private function manageEvent($eventParams = array())
...
-            $this->to = $event->get('to');
+            /* Set default as array to avoid the to to null and broke on PHP8 */
+            $this->to = $event->get('to', array());
         }
         if (empty($updateDisable['bounce'])) {
-            $this->Sender = $event->get('bounce');
+            $this->Sender = $event->get('bounce', '');
         }
         $this->eventMessage = $event->get('message');
+        /* Need loose compare (default to true) */
```
Comment (olleharstedt) [inline application/core/LimeMailer.php:538]: Why does it need lose compare?

Comment (olleharstedt) [reply application/core/LimeMailer.php:538]: OK, can you add that motivation in the comment, please? Alternatively use `=== "" || === 0`.

Comment (Shnoulle) [reply application/core/LimeMailer.php:538]: Next time i don't add comment … it was loose compare before …

Comment (Shnoulle) [reply application/core/LimeMailer.php]:
```
        /* Need loose compare : if null, return true (default) */
        /* Plugin can send anything for false : don't break API by move to strict compare */
```

Comment (olleharstedt) [reply application/core/LimeMailer.php]: Don't be grumpy :sweat_smile: Thanks.

Comment (Shnoulle) [reply application/core/LimeMailer.php]: > Don't be grumpy 

I can not : i'm french ;) (and  old)

**Pull request merged**